apollo-federation/src/connectors/json_selection/methods/public/lte.rs

```diff
@@ -0,0 +1,372 @@
+    let matches = value_opt.is_some_and(|value| {
+        match (data, &value) {
+            // Number comparisons
+            (JSON::Number(left), JSON::Number(right)) => {
+                left.as_f64().unwrap_or(0.0) <= right.as_f64().unwrap_or(0.0)
+            }
+            // String comparisons
+            (JSON::String(left), JSON::String(right)) => left <= right,
+            // Boolean comparisons
+            (JSON::Bool(left), JSON::Bool(right)) => left <= right,
+            // Null comparisons (null == null)
+            (JSON::Null, JSON::Null) => true,
+            // Mixed types or uncomparable types (including arrays and objects) return false
+            _ => {
+                apply_to_errors.push(ApplyToError::new(
+                    format!(
+                        "Method ->{} can directly compare numbers, strings, booleans, and null. Either a mix of these was provided or something else such as an array or object. Found: {} <= {}",
+                        method_name.as_ref(),
+                        data,
+                        value
+                    ),
+                    input_path.to_vec(),
+                    method_name.range(),
+                ));
+
+                false
+            }
+        }
+    });
+
+    (Some(JSON::Bool(matches)), apply_to_errors)
+}
+
+#[allow(dead_code)] // method type-checking disabled until we add name resolution
+fn lte_shape(
+    method_name: &WithRange<String>,
+    method_args: Option<&MethodArgs>,
+    input_shape: Shape,
+    dollar_shape: Shape,
+    named_var_shapes: &IndexMap<&str, Shape>,
+    source_id: &SourceId,
+) -> Shape {
+    let arg_count = method_args.map(|args| args.args.len()).unwrap_or_default();
+    if arg_count > 1 {
+        return Shape::error(
+            format!(
+                "Method ->{} requires only one argument, but {arg_count} were provided",
+                method_name.as_ref(),
+            ),
+            vec![],
+        );
+    }
+
+    let Some(first_arg) = method_args.and_then(|args| args.args.first()) else {
+        return Shape::error(
+            format!("Method ->{} requires one argument", method_name.as_ref()),
+            method_name.shape_location(source_id),
+        );
+    };
+
+    let arg_shape = first_arg.compute_output_shape(
+        input_shape.clone(),
+        dollar_shape,
+        named_var_shapes,
+        source_id,
+    );
+
+    match (arg_shape.case(), input_shape.case()) {
+        (ShapeCase::Unknown, ShapeCase::Unknown)
+        | (ShapeCase::Int(_), ShapeCase::Int(_))
+        | (ShapeCase::Float, ShapeCase::Float)
+        | (ShapeCase::Int(_), ShapeCase::Float)
+        | (ShapeCase::Float, ShapeCase::Int(_))
+        | (ShapeCase::Bool(_), ShapeCase::Bool(_))
+        | (ShapeCase::Null, ShapeCase::Null)
+        | (ShapeCase::String(_), ShapeCase::String(_)) => {
+            Shape::bool(method_name.shape_location(source_id))
+        }
+        _ => Shape::error(
```

Comment:
This makes it an error if the input shape is `ShapeCase::Name`, which could become/represent a number at runtime. Instead of lumping that case into the `_ => Shape::error` branch, let's make it explicit.